File: src/src_core/fmg/core/img/MosaicGroupController.java
package fmg.core.img;

import fmg.core.types.EMosaicGroup;

/**
 * MVC controller of {@link EMosaicGroup} image
 *
 * @param <TImage> plaform specific view/image/picture or other display context/canvas/window/panel
 * @param <TImageView> MVC view
 */
public abstract class MosaicGroupController< TImage,
                                              TImageView extends WithBurgerMenuView<TImage, MosaicGroupModel>>
                extends AnimatedImgController<TImage, TImageView, MosaicGroupModel>
{

   protected MosaicGroupController(boolean showBurgerMenu, TImageView imageView) {
      super(imageView);

      getView().getBurgerMenuModel().setShow(showBurgerMenu);

      addModelTransformer(new MosaicGroupTransformer());
      usePolarLightFgTransforming(true);
      useRotateTransforming(true);
   }

}

File: src/src_core/fmg/core/img/MosaicGroupController.cs
namespace fmg.core.img {

   /// <summary> MVC controller of <see cref="EMosaicGroup"/> image </summary>
   /// <typeparam name="TImage">platform specific view/image/picture or other display context/canvas/window/panel</typeparam>
   /// <typeparam name="MosaicGroupModel">MVC view</typeparam>
   public abstract class MosaicGroupController<TImage, TImageView>
                       : AnimatedImgController<TImage, TImageView, MosaicGroupModel>
      where TImage : class
      where TImageView : WithBurgerMenuView<TImage, MosaicGroupModel>
   {

      protected MosaicGroupController(bool showBurgerMenu, TImageView imageView)
         : base(imageView)
      {
         View.BurgerMenuModel.Show = showBurgerMenu;

         AddModelTransformer(new MosaicGroupTransformer());
         UsePolarLightFgTransforming(true);
         UseRotateTransforming(true);
      }

   }

}
